### DISCUSSION TITLE
plugin - how to prevent json interpretation

### DISCUSSION THREAD

**Discussion by mikeTWC1984** [jhuckaby/Cronicle#509] [Q&A]
When you write custom plugin, and produce some json output job execution engine will interpret it and exclude from logging. What is the best to prevent that? In other words how to set up "Interpret JSON" check box on simple custom plugin. 
For example for plugin like this:
```javascript
#!/usr/bin/env node

console.log(JSON.stringify({foo: "bar"}))
```

**Comment by jhuckaby** [comment]
This is resolved in Cronicle v2, which has a redesigned STDIO interface with Plugins.  But for the current version, the engine looks for anything that is a valid JSON document on a single line (basically this regex: `/^\{.+\}$/`), and interprets that (and excludes it from logging).  The best way to get around this with Cronicle v0 is to log a prefix or suffix that makes it clear that the line is not JSON, so it doesn't match the regex.  Like this:

```js
console.log( "LOG: " + JSON.stringify({foo: "bar"}))
```

Or log your JSON with pretty-printing, so it spans multiple lines:

```js
console.log( JSON.stringify({foo: "bar"}, null, "\t") )
```

**Reply by mikeTWC1984** [reply]
Ha, that simple, thanks.

Another quirk I came across while creating template - there is a "inline variable" feature for plugin parameters, say if parameter value is "$PATH/some/path", then $PATH is replaced with current env variable.
The problem here is sometime I just need that $ sign to be part of the value, and looks like there is no way to escape/avoid this interpolation.
Say my parameter should contain bash command, e.g. "export A=10; echo $A". In this case $A will be replaced with '', and it will brake command.

**Reply by mikeTWC1984** [reply]
https://github.com/jhuckaby/Cronicle/blob/master/lib/job.js#L512:L520

**Reply by jhuckaby** [reply]
Got it, thanks Mike!  I'll make this behavior optional, or perhaps escapable.

**Reply by jhuckaby** [reply]
You can now escape the shell variables in v0.9.8.  Just precede the dollar sign with a backslash, e.g. `\$FOO`.

https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.8
